<!-- -*- mode: XML -*- -->

<launch>

<!-- PWM DRIVER NODES -->  

  <include ns="steering" file="$(find pwm_sysfs_driver)/launch/pwm_sysfs_driver_node.launch" >
  	<arg name="pwm_sysfs_dir" value="/sys/class/pwm/pwmchip0/pwm0" />
  	<arg name="initial_enable" value="true" />
  	<arg name="initial_invert_polarity" value="false" />
  	<arg name="initial_period" value="5000000" />
  	<arg name="initial_duty" value=".15" />  
  </include>
  
  <include ns="throttle" file="$(find pwm_sysfs_driver)/launch/pwm_sysfs_driver_node.launch" >
  	<arg name="pwm_sysfs_dir" value="/sys/class/pwm/pwmchip0/pwm2" />
  	<arg name="initial_enable" value="true" />
  	<arg name="initial_invert_polarity" value="false" />
  	<arg name="initial_period" value="5000000" />
  	<arg name="initial_duty" value=".15" />
  </include>

	<remap from="steering/duty" to="steering/pwm_sysfs_driver_node/duty" />
  <remap from="steering/enable" to="steering/pwm_sysfs_driver_node/enable" />
  <remap from="steering/invert_polarity" to="steering/pwm_sysfs_driver_node/invert_polarity" /> 
  <remap from="steering/period" to="steering/pwm_sysfs_driver_node/period" />
  	
  <remap from="throttle/duty" to="throttle/pwm_sysfs_driver_node/duty" />
  <remap from="throttle/enable" to="throttle/pwm_sysfs_driver_node/enable" />
  <remap from="throttle/invert_polarity" to="throttle/pwm_sysfs_driver_node/invert_polarity" /> 
  <remap from="throttle/period" to="throttle/pwm_sysfs_driver_node/period" />

  <node pkg="twist_to_pwm" type="twist_to_pwm.py" name="twist_to_pwm"/>


<!-- TELEOPERATION NODES -->  
  <node pkg="joy" type="joy_node" name="joy_node"/>
  <node pkg="ack_teleop" type="ack_teleop.py" name="ack_teleop"/>
  
  
<!-- SENSOR NODES -->
  <node pkg="urg_node" type="urg_node" name="urg_node"/>
  <include file="$(find razor_imu_9dof)/launch/razor-pub.launch" />
 	<node pkg="pointgrey_camera_driver" type="camera_node" name="camera_node"/>
 
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser"
  	args="0.0 0.0 0.0 0.0 0.0 0.0 /base_link /laser 40" />
  <node pkg="tf" type="static_transform_publisher" name="base_imu_link"
  	args="0.0 0.0 0.0 0.0 0.0 0.0 /base_link /imu 40" />
  		    
  <remap from="imu/data" to="imu" /> <!-- fix discrepancy -->
  
<!-- LASER SCAN MATCHER -->
<!--
  <node pkg="laser_scan_matcher" type="laser_scan_matcher_node" name="laser_scan_matcher_node" >
  	<param name="fixed_frame" value="odom" />
		<param name="use_imu" value="true" />
  	<param name="use_odom" value="false" />
  	<param name="use_alpha_beta" value="false" />
		<param name="max_iterations" value="20" />
		<param name="kf_dist_linear" value="0.5" />
		<param name="kf_dist_angular" value="1.0" />	

  	<param name="publish_tf" value="true" />
  	<param name="publish_pose" value="true" />
  </node>
  -->
  
<!-- HECTOR_SLAM [NON-FUNCTIONAL]-->
	<node pkg="hector_mapping" type="hector_mapping" name="hector_mapping">
		<param name="pub_map_odom_transform" value="true"/>
		<param name="pub_map_scanmatch_transform" value="false"/>
		<param name="base_frame" value="base_link"/>
		<param name="odom_frame" value="base_link"/>
	</node>
	<include file="$(find hector_geotiff)/launch/geotiff_mapper.launch"/>

<!-- GMAPPING -->
<!--
	<node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" >
		<param name="map_udpate_interval" value="1.0"/>
		<param name="maxUrange" value="4.5"/>
		<param name="sigma" value="0.1"/>
		<param name="kernelSize" value="1"/>
		<param name="lstep" value="0.15"/>
		<param name="astep" value="0.15"/>
		<param name="iterations" value="5"/>
		<param name="lsigma" value="0.1"/>
		<param name="ogain" value="3.0"/>
		<param name="lskip" value="1"/>
		<param name="srr" value="0.1"/>
		<param name="srt" value="0.2"/>
		<param name="str" value="0.1"/>
		<param name="stt" value="0.2"/>
		<param name="linearUpdate" value="0.3"/>
		<param name="angularUpdate" value="0.3"/>
		<param name="temporalUpdate" value="0.5"/>
		<param name="resampleThreshold" value="0.5"/>
		<param name="particles" value="50"/>
		<param name="xmin" value="-20.0"/>
		<param name="ymin" value="-20.0"/>
		<param name="xmax" value="20.0"/>
		<param name="ymax" value="20.0"/>
		<param name="delta" value="0.02"/>
		<param name="llsamplerange" value="0.01"/>
		<param name="llsamplestep" value="0.05"/>
		<param name="lasamplerange" value="0.05"/>
		<param name="lasamplestep" value="0.05"/>
	</node>
	-->
</launch>
